parsl/executors/high_throughput/process_worker_pool.py

```diff
@@ -731,9 +731,19 @@ def worker(
         os.sched_setaffinity(0, my_cores)  # type: ignore[attr-defined, unused-ignore]
         logger.info("Set worker CPU affinity to {}".format(my_cores))
 
+    # If CUDA devices, find total number of devices to allow for MPS
+    # See: https://developer.nvidia.com/system-management-interface
+    try:
+        mod_factor = int(os.popen("nvidia-smi -L | wc -l").read())
```

Comment:
probably should be calling .close() on the popened object, after its been read()